src/Shared/NamedPipeUtil.cs

```diff
@@ -25,5 +27,71 @@ internal static string GetPipeNameOrPath(string pipeName)
                 return pipeName;
             }
         }
+        internal static NamedPipeClientStream GetClientStream(string pipeName, int timeout)
+        {
+            // Try and connect to the process.
+            var nodeStream = new NamedPipeClientStream(".", pipeName, PipeDirection.InOut, PipeOptions.Asynchronous
```

Comment:
is this disposable?

Reply:
Yes it is. In case of some error is the stream disposed inside this method, else it is up to user to dispose it.